newlib/libc/include/sys/config.h

```diff
@@ -195,7 +195,6 @@
 
 #ifdef __XTENSA__
 #include <xtensa/config/core-isa.h>
-#define MALLOC_ALIGNMENT ((XCHAL_DATA_WIDTH) < 16 ? 16 : (XCHAL_DATA_WIDTH))
```

Comment:
Hmm, looks like imx sets XCHAL_DATA_WIDTH to 16.  So wonder if this should be
`#define MALLOC_ALIGNMENT XCHAL_DATA_WIDTH`

Guessing that will still cause issues on xtensa for i.mx?

Reply:
> Guessing that will still cause issues on xtensa for i.mx?

@galak Yes, this issue will persist as long as `MALLOC_ALIGNMENT` is 16 (and maybe others except 8 too) since we are not fixing the root cause here.

While it would be nice to fix the malloc itself, I do not have bandwidth to debug and fix it at the moment.

As for the i.MX with `XCHAL_DATA_WIDTH` of 16, I do not think there will be any issues using the `MALLOC_ALIGNMENT` of 8 other than some minor performance penalties for the conventional un-optimised code.